pkg/chaincode/example_test.go

```diff
@@ -77,6 +84,14 @@ func newGrpcConnection() *grpc.ClientConn {
 	return connection
 }
 
+func self[T any](v T) T {
+	return v
+}
+
+func newGateway(connection grpc.ClientConnInterface, id identity.SigningIdentity) (*client.Gateway, error) {
+	return client.Connect(id, client.WithClientConnection(connection), client.WithHash(self[[]byte]), client.WithSign(id.Sign))
+}
+
```

Comment:
This is an example of how to use the API, and part of the documentation. Don't make the user of the API do all this extra work.